app/core/pokemon-entity.ts

```diff
@@ -1404,7 +1426,7 @@ export class PokemonEntity extends Schema implements IPokemonEntity {
       target.status.triggerProtect(2000)
       target.handleHeal(20, target, 0, false)
       target.items.delete(Item.BABIRI_BERRY)
-      target.refToBoardPokemon.items.delete(Item.BABIRI_BERRY)
+      target.addPermanentItem(Item.BABIRI_BERRY, true)
```

Comment:
the `removePermanentItem` method could handle removing the item both from the entity and the board pokémon